### ISSUE TITLE
The default value for struct Rational behaves oddly

### ISSUE THREAD

**Issue by dmytro-sheyko** [tompazourek/Rationals#24]
The default value is both `IsZero` and `IsOne` at the same time and equals to any `Rational` value.
```
            Rational weird = default(Rational); // new Rational();
            Console.WriteLine($"{weird}.IsZero : {weird.IsZero}"); // True
            Console.WriteLine($"{weird}.IsOne  : {weird.IsOne}");  // True
            Console.WriteLine($"{weird} == 10  : {weird == 10}");  // True
            Console.WriteLine($"10 == {weird}  : {10 == weird}");  // True
```
Maybe it's worth to consider `(0/0)` form as `NaN`? Or at least make it behave as `Zero` (i.e. `(0/1)`) consistently.

**Comment by tompazourek** [comment]
Good point. You're right that it's a bit weird.

I'm not 100% decided on which way to handle this. The NaN arithmetic can result in all kinds of strange situations as well and feels a bit strange to have NaN as the default. On the other hand, it might be the most straightforward and expected way to deal with this issue.

I'll try to draft some proof-of-concept of how it could work.

Thanks for bringing this up! 😊

**Comment by tompazourek** [comment]
I went with the #26 solution. This is now released as version 2.0.0.

@dmytro-sheyko Thank you a lot again for your contribution related to this. Any more improvements or ideas are welcome.
